src/content/PagePartials/index.ts

```diff
@@ -0,0 +1,76 @@
+import { defineCollection, z } from 'astro:content';
+import {
+  PagePartialCollectionEntry as query,
+  type PagePartialCollectionEntryQuery,
+  type SiteLocale
+} from '@lib/datocms/types';
+import { datocmsCollection, datocmsRequest } from '@lib/datocms';
+import { combine } from '@lib/content';
+
+export type PagePartialCollectionEntry = PagePartialCollectionEntryQuery['entry'] & {
+  recordId: string,
+  id: string,
+  locale: SiteLocale,
+};
+
+const name = 'PagePartials' as const;
+
+const loadEntry = async (id: string, locale?: SiteLocale | null) => {
+  if (!locale) {
+    return;
+  }
+  const { entry } = await datocmsRequest<PagePartialCollectionEntryQuery>({ query, variables: { id, locale } });
+  return {
+    ...entry,
+    recordId: entry.id,
+    id: combine({ id: entry.id, locale }),
+    locale,
+  } satisfies PagePartialCollectionEntry;
+};
+
+/** 
+ * Loads all entries from the collection, mapping them to their respective locales.
+ * 
+ * @returns A promise that resolves to an array of PagePartialCollectionEntry objects.
+ **/
+const loadCollection = async () => {
+  const items = (await datocmsCollection<{ 
+    id: string, 
+    _allBlocksLocales: { locale: SiteLocale }[] 
+  }>({
+    collection: name,
+    fragment: /* graphql */`
+      id
+      # We want all locales for each entry. Since the blocks are localized, 
+      # we can fetch the locales via _allBlocksLocales
+      _allBlocksLocales {
+        locale
+      }
+    `,
+  }))
+    // Flatten the array of entries to get an array of { id, locale } pairs.
+    .flatMap(({ id, _allBlocksLocales }) => _allBlocksLocales
+      .map(({ locale }) => ({ id, locale })) 
+    );
+
+  // For each id/locale pair, load the entry and return it.
+  // Note that this might be slow if there are many entries, as it makes a
+  // separate request for each entry.
```

Comment:
What do you think of using grahpql query batching here? tested it and it seems faster (don't know if there are limits on how much queries you can batch, but might be worth looking into):

With:
![image](https://github.com/user-attachments/assets/78f627ce-4ba2-42d4-a7a0-4e2b70c6c305)

Without
![image](https://github.com/user-attachments/assets/236c4937-c06d-4c05-8e9d-b0ec57e6ee1e)


Not that much of a difference, but can imagine that this will improve things with a large number of queries

Reply:
Will keep this in mind for the next interation, thanks! :)